keras_cv/layers/preprocessing3d/global_random_translation.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d import base_augmentation_layer_3d
+from keras_cv.ops.point_cloud import coordinate_transform
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+
+
+class GlobalRandomTranslation(base_augmentation_layer_3d.BaseAugmentationLayer3D):
+    """A preprocessing layer which randomly translates point clouds and bounding boxes along
+    X, Y, and Z axes during training.
+
+    This layer will randomly translate the whole scene along the  X, Y, and Z axes based on three randomly sampled
+    translation factors following three normal distributions centered at 0 with standard deviation  [x_translation_stddev, y_translation_stddev, z_translation_stddev].
+    During inference time, the output will be identical to input. Call the layer with `training=True` to translate the input.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 7 features are [x, y, z, dx, dy, dz, phi].
+
+    Output shape:
+      A tuple of two Tensors (point_clouds, bounding_boxes) with the same shape as input Tensors.
+
+    Arguments:
+      x_translation_stddev: A float scaler or Tensor sets the translation noise standard deviation along the X axis.
```

Comment:
will there be a lower bound and upper bound to translation? In 2D we usually have `height_factor` and `width_factor` as fraction of image shape, in 3D, is there a maximum distance from sensor to the farther-est point?

Reply:
Unlike 2D, 3D is open space. We can truncate distance in the voxelization layer.